core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -882,16 +932,21 @@ object Chunk {
    */
   private[zio] def classTagOf[A](chunk: Chunk[A]): ClassTag[A] =
     chunk match {
-      case x: Arr[A]         => x.classTag
-      case x: Concat[A]      => x.classTag
-      case Empty             => classTag[java.lang.Object].asInstanceOf[ClassTag[A]]
-      case x: Singleton[A]   => x.classTag
-      case x: Slice[A]       => x.classTag
-      case x: VectorChunk[A] => x.classTag
-      case _: BitChunk       => ClassTag.Boolean.asInstanceOf[ClassTag[A]]
+      case x: Arr[A]          => x.classTag
+      case x: ArraySupport[A] => x.classTag
+      case x: Concat[A]       => x.classTag
+      case Empty              => classTag[java.lang.Object].asInstanceOf[ClassTag[A]]
+      case x: Singleton[A]    => x.classTag
+      case x: Slice[A]        => x.classTag
+      case x: VectorChunk[A]  => x.classTag
+      case _: BitChunk        => ClassTag.Boolean.asInstanceOf[ClassTag[A]]
     }
 
-  private final case class Arr[A](private val array: Array[A]) extends Chunk[A] with Serializable { self =>
+  private final case class Arr[A](array: Array[A]) extends ArraySupport[A]
+
+  private[zio] trait ArraySupport[A] extends Chunk[A] with Serializable { self =>
```

Comment:
To make this simpler, instead of a mixin trait, let's make `Arr[A]` an abstract class and have all the primitive subtypes inherit it. In addition, we can add a `RefChunkArr` subtype for `A <: AnyRef` types, which will subsume the role of `Arr[A]` today.